packages/cli/generators/relation/templates/controller-relation-template-has-one.ts.ejs

```diff
@@ -0,0 +1,110 @@
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.path.<%= sourceModelPrimaryKeyType %>('id') id: <%= sourceModelPrimaryKeyType %>,
+    @param.query.object('filter') filter?: Filter<<%= targetModelClassName %>>,
+  ): Promise<<%= targetModelClassName %>[]> {
+    return this.<%= paramSourceRepository %>.<%= relationPropertyName %>(id).find(filter);
+  }
+
+  @post('/<%= sourceModelPath %>/{id}/<%= targetModelPath %>', {
+    responses: {
+      '200': {
+        description: '<%= sourceModelClassName %> model instance',
+        content: {'application/json': {schema: getModelSchemaRef(<%= targetModelClassName %>)}},
+      },
+    },
+  })
+  async create(
+    @param.path.<%= sourceModelPrimaryKeyType %>('id') id: typeof <%= sourceModelClassName %>.prototype.<%= sourceModelPrimaryKey %>,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(<%= targetModelClassName %>, {
+            title: 'New<%= targetModelClassName %>In<%= sourceModelClassName %>',
+            exclude: ['<%= targetModelPrimaryKey %>'],
+            optional: ['<%= foreignKeyName %>']
+          }),
+        },
+      },
+    }) <%= targetModelRequestBody %>: Omit<<%= targetModelClassName %>, '<%= targetModelPrimaryKey %>'>,
+  ): Promise<<%= targetModelClassName %>> {
+    return this.<%= paramSourceRepository %>.<%= relationPropertyName %>(id).create(<%= targetModelRequestBody %>);
+  }
+
+  @patch('/<%= sourceModelPath %>/{id}/<%= targetModelPath %>', {
+    responses: {
+      '200': {
+        description: '<%= sourceModelClassName %>.<%= targetModelClassName %> PATCH success count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async patch(
+    @param.path.<%= sourceModelPrimaryKeyType %>('id') id: <%= sourceModelPrimaryKeyType %>,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(<%= targetModelClassName %>, {partial: true}),
+        },
+      },
+    })
+    <%= targetModelRequestBody %>: Partial<<%= targetModelClassName %>>,
+    @param.query.object('where', getWhereSchemaFor(<%= targetModelClassName %>)) where?: Where<<%= targetModelClassName %>>,
+  ): Promise<Count> {
+    return this.<%= paramSourceRepository %>.<%= relationPropertyName %>(id).patch(<%= targetModelRequestBody %>, where);
+  }
+
+  @del('/<%= sourceModelPath %>/{id}/<%= targetModelPath %>', {
+    responses: {
+      '200': {
+        description: '<%= sourceModelClassName %>.<%= targetModelClassName %> DELETE success count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async delete(
+    @param.path.<%= sourceModelPrimaryKeyType %>('id') id: <%= sourceModelPrimaryKeyType %>,
+    @param.query.object('where', getWhereSchemaFor(<%= targetModelClassName %>)) where?: Where<<%= targetModelClassName %>>,
+  ): Promise<Count> {
```

Comment:
@agnes512  I am not sure if I should be using Count here or something else like Bool because the count would always be either 0 or 1

Reply:
Agreed. I think we can leave it out.